test/uniqBy.js

```diff
@@ -0,0 +1,39 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('uniqBy', function() {
+
+  var objs = [
+    {i: 1, v: 1}, {i: 2, v: 2}, {i: 3, v: 3}, {i: 4, v: 4}, {i: 5, v: 5},
+    {i: 6, v: 6}, {i: 7, v: 7}, {i: 8, v: 8}, {i: 9, v: 9}, {i: 10, v: 10}
+  ];
+
+  var objs2 = [
+    {i: 1, v: 1}, {i: 2, v: 2}, {i: 2, v: 3}, {i: 4, v: 4}, {i: 5, v: 5},
+    {i: 6, v: 6}, {i: 6, v: 7}, {i: 6, v: 8}, {i: 9, v: 9}, {i: 5, v: 10}
+  ];
+
+  var getI = R.prop('i');
+
+  it('returns a set from any array (i.e. purges duplicate elements) based on predicate', function() {
+    assert.deepEqual(R.uniqBy(getI, objs), objs);
+    console.log(R.uniqBy(getI, objs2));
+    assert.deepEqual(R.uniqBy(getI, objs2), [{i: 1, v: 1}, {i: 2, v: 2}, {i: 4, v: 4}, {i: 5, v: 5}, {i: 6, v: 6}, {i: 9, v: 9}]);
+  });
+
+  it('keeps elements from the left', function() {
+    assert.deepEqual(R.uniqBy(getI, [{i: 1}, {i: 2}, {i: 3}, {i: 4}, {i: 1}]), [{i: 1}, {i: 2}, {i: 3}, {i: 4}]);
+  });
+
+  it('returns an empty array for an empty array', function() {
+    assert.deepEqual(R.uniqBy(getI, []), []);
+  });
+
+  it('is curried', function() {
+    assert.strictEqual(typeof R.uniqBy(getI), 'function');
+    assert.deepEqual(R.uniqBy(getI)(objs), objs);
+    assert.deepEqual(R.uniqBy(getI)(objs2), [{i: 1, v: 1}, {i: 2, v: 2}, {i: 4, v: 4}, {i: 5, v: 5}, {i: 6, v: 6}, {i: 9, v: 9}]);
+  });
```

Comment:
It's no longer necessary to assert that each function we add to the library is curried, as we assert this elsewhere for *all* Ramda functions. You could remove this block.